lib/queue.rb

```diff
@@ -1,28 +1,27 @@
 class Queue
```

Comment:
This works, but it's not implemented with a circular buffer.  I suggest you review the notes on circular buffers and just verify that you understand how it works.